components/camel-git/src/main/java/org/apache/camel/component/git/producer/GitProducer.java

```diff
@@ -407,6 +413,28 @@ protected void doPush(Exchange exchange, String operation) throws Exception {
         updateExchange(exchange, result);
     }
 
+    protected void doPushTag(Exchange exchange, String operation) throws Exception {
+        Iterable<PushResult> result = null;
+        try {
+            if (ObjectHelper.isEmpty(endpoint.getRemoteName())) {
+                throw new IllegalArgumentException("Remote name must be specified to execute " + operation);
+            }
+            if (ObjectHelper.isEmpty(endpoint.getTagName())) {
+                throw new IllegalArgumentException("Tag Name must be specified to execute " + operation);
+            }
+            if (ObjectHelper.isNotEmpty(endpoint.getUsername()) && ObjectHelper.isNotEmpty(endpoint.getPassword())) {
+                UsernamePasswordCredentialsProvider credentials = new UsernamePasswordCredentialsProvider(endpoint.getUsername(), endpoint.getPassword());
+                result = git.push().setCredentialsProvider(credentials).setRemote(endpoint.getRemoteName()).add(Constants.R_TAGS + endpoint.getTagName()).call();
+            } else {
+                result = git.push().setRemote(endpoint.getRemoteName()).add(Constants.R_TAGS + endpoint.getTagName()).call();
+            }
+        } catch (Exception e) {
+            LOG.error("There was an error in Git " + operation + " operation");
```

Comment:
It would be better to use LOG.error("There was an error in Git {} operation", operation); instead

Reply:
Sure! But it's was followed others methods: i do not want to have modified the current normalization form. Do i change this, anyway?

Reply:
We can extract the common part in an util method, but I would leave the operations separated